pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -328,9 +339,14 @@ func dropMachineSelectorFile(file rkev1.RKEProvisioningFiles, cluster *v1.Cluste
 }
 
 // MachineSelectorFileExists returns a boolean to indicate if the provided RKEProvisioningFiles exist in the provided cluster.
-func MachineSelectorFileExists(file rkev1.RKEProvisioningFiles, cluster *v1.Cluster) bool {
+func MachineSelectorFileExists(file rkev1.RKEProvisioningFiles, cluster *v1.Cluster, ignoreValueCheck bool) bool {
 	for _, item := range cluster.Spec.RKEConfig.MachineSelectorFiles {
-		if equality.Semantic.DeepEqual(file, item) {
+		fromCluster := item.DeepCopy()
+		if ignoreValueCheck {
+			cleanupHash(fromCluster)
+			cleanupHash(&file)
```

Comment:
`cleanupHash(&file)` Only needs to be called once so this could be moved out of the loop.

Reply:
fixed